### PR TITLE
Update Console options for .NET Core Console build

### PR DESCRIPTION
**Pull request by ChrisMaddock**
Relates to #475. This PR updates the console options for the .NET Core build, disabling those which won't initially be available. (With a helpful error message, rather than just removing them from the build!)

It also does a fair amount of refactoring at the same time - I think this will be ok to review commit-by-commit, but shout if you'd prefer I separate the refactoring out. 👍 

Key things that were refactored (which resulted in a log of knock on effects):
- Combining all console options into a single file - previously they were still separated into two files, from when this lived in the same repo as NUnitLite.
- Removing a ConsoleOptions constructor, which was only used by tests. (Resulted in lots of changes to test files - replaced with an NSubstitute alternative)

### CODE REVIEW

File: src/NUnitConsole/nunit3-console.tests/ConsoleMocks.cs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ﻿// ***********************************************************************
```
Comment (ChrisMaddock) [inline src/NUnitConsole/nunit3-console.tests/ConsoleMocks.cs]: This is not a file move, GitHub has matched these up incorrectly.

`IConverter` was an unused file in the console project - unrelated to this PR. ConsoleMocks is a utility class to create a mock ConsoleOptions object in the test suite.

File: src/NUnitConsole/nunit3-console/OptionsUtils/OptionParser.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
+﻿// ***********************************************************************
```
Comment (ChrisMaddock) [inline src/NUnitConsole/nunit3-console/OptionsUtils/OptionParser.cs]: All this code is just moved out of the original two CommandLineOptions classes, to try and limit the file size.

File: src/NUnitConsole/nunit3-console.tests/ConsoleMocks.cs (hunk 1 above)
Comment (ChrisMaddock) [inline src/NUnitConsole/nunit3-console.tests/ConsoleMocks.cs]: This isn't a file move, GitHub has matched these two files up incorrectly.
- IConverter.cs was an unused class in the console
- ConsoleMocks.cs has been added to move test-only code out of the production project and into the test suite.

**Comment by ChrisMaddock** [comment]
Rebased now the other PR has been merged - would appreciate any reviews. Thanks! 😄

**Comment by ChrisMaddock** [comment]
Hey @nunit/engine-team - this has been sat here for a couple of months waiting review, and I'd like to get it merged in soon for the next release. Appreicate everyone's busy at the moment - so I'm going to break it into three chunks, and merge the simple refactoring changes straight in. It would be great if someone could take a look at the actual code changes however - I'll hopefully get that sorted today. 🙂

**Comment by ChrisMaddock** [comment]
Ugh - I didn't intend to merge this, and I'm not totally clear on what happened. :-/ I think while cherry-picking for my other branches, I must have accidentally merged my local branch into master somehow and pushed that. I thought the master branch protection would have stopped me doing something like that - but perhaps it doesn't work if all CI etc has already passed for the commit...

Given there's no easy way for me to unpick this, and the fact I was going to merge most of this directly anyway, I'll leave it as is this time. Still welcome any comments on this PR - happy to act on the retrospectively. 👍

**Comment by CharliePoole** [comment]
Retrospectively... it looks good! :smiley:

**Comment by ChrisMaddock** [comment]
Thanks for checking, Charlie! 😅

File: src/NUnitConsole/nunit3-console.tests/MakeTestPackageTests.cs [hunk 3]
```diff
@@ -53,8 +52,17 @@ public void MultipleAssemblies()
...
         [TestCase("--dispose-runners", "DisposeRunners", true)]
+        [TestCase("--config=Release", "ActiveConfig", "Release")]
+        [TestCase("--trace=Error", "InternalTraceLevel", "Error")]
+        [TestCase("--trace=error", "InternalTraceLevel", "Error")]
+        [TestCase("--seed=1234", "RandomSeed", 1234)]
+        [TestCase("--workers=3", "NumberOfTestWorkers", 3)]
+        [TestCase("--workers=0", "NumberOfTestWorkers", 0)]
+        [TestCase("--params:X=5;Y=7", "TestParameters", "X=5;Y=7")]
+        [TestCase("--skipnontestassemblies", "SkipNonTestAssemblies", true)]
+#if NET35
```
Comment (mikkelbu) [inline src/NUnitConsole/nunit3-console.tests/MakeTestPackageTests.cs]: In the production code we use `#if NET20`

File: src/NUnitConsole/nunit3-console.tests/MakeTestPackageTests.cs [hunk 4]
```diff
@@ -64,35 +72,29 @@ public void MultipleAssemblies()
...
         {
-            var options = new ConsoleOptions("test.dll", option);
+            var options = ConsoleMocks.Options("test.dll", option);
             var package = ConsoleRunner.MakeTestPackage(options);
 
             Assert.That(package.Settings.ContainsKey(key), "Setting not included for {0}", option);
             Assert.That(package.Settings[key], Is.EqualTo(val), "NumberOfTestWorkers not set correctly for {0}", option);
         }
 
+#if NET35
```
Comment (mikkelbu) [inline src/NUnitConsole/nunit3-console.tests/MakeTestPackageTests.cs]: See above comment regarding `#if NET20`

File: src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs [hunk 5]
```diff
@@ -163,14 +162,16 @@ public void NoInputFiles()
         [TestCase("StopOnError", "stoponerror")]
         [TestCase("WaitBeforeExit", "wait")]
         [TestCase("NoHeader", "noheader|noh")]
-        [TestCase("RunAsX86", "x86")]
         [TestCase("DisposeRunners", "dispose-runners")]
-        [TestCase("ShadowCopyFiles", "shadowcopy")]
         [TestCase("TeamCity", "teamcity")]
+        [TestCase("SkipNonTestAssemblies", "skipnontestassemblies")]
+#if NET35
```
Comment (mikkelbu) [inline src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs]: In the production code we use `#if NET20`

File: src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs [hunk 6]
```diff
@@ -187,39 +188,41 @@ public void CanRecognizeBooleanOptions(string propertyName, string pattern)
...
-        [TestCase("DomainUsage", "domain", new string[] { "None", "Single", "Multiple" }, new string[] { "JUNK" })]
-        [TestCase("Framework", "framework", new string[] { "net-4.0" }, new string[0])]
         [TestCase("OutFile", "output|out", new string[] { "output.txt" }, new string[0])]
-        [TestCase("ConfigurationFile", "configfile", new string[] { "mytest.config" }, new string[0] )]
         [TestCase("WorkDirectory", "work", new string[] { "results" }, new string[0])]
         [TestCase("DisplayTestLabels", "labels", new string[] { "Off", "On", "Before", "After", "All" }, new string[] { "JUNK" })]
         [TestCase("InternalTraceLevel", "trace", new string[] { "Off", "Error", "Warning", "Info", "Debug", "Verbose" }, new string[] { "JUNK" })]
         [TestCase("DefaultTestNamePattern", "test-name-format", new string[] { "{m}{a}" }, new string[0])]
         [TestCase("ConsoleEncoding", "encoding", new string[] { "utf-8", "ascii", "unicode" }, new string[0])]
+#if NET35
```
Comment (mikkelbu) [inline src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs]: Same comment as above

File: src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs [hunk 7]
```diff
@@ -232,30 +235,34 @@ public void CanRecognizeStringOptions(string propertyName, string pattern, strin
...
                 {
                     string optionPlusValue = string.Format("--{0}:{1}", option, value);
-                    ConsoleOptions options = new ConsoleOptions(optionPlusValue);
+                    ConsoleOptions options = ConsoleMocks.Options(optionPlusValue);
                     Assert.That(options.Validate(), Is.False, "Should not be valid: " + optionPlusValue);
                 }
             }
         }
 
+#if NET35
```
Comment (mikkelbu) [inline src/NUnitConsole/nunit3-console.tests/CommandLineTests.cs]: Same comment as above

**Comment by ChrisMaddock** [comment]
Thanks Mikkel! I'll get back to these at the weekend. 😄

**Pull request merged**